dashboard/modules/job/common.py

```diff
@@ -189,13 +189,28 @@ def __init__(self, gcs_aio_client: GcsAioClient):
         self._gcs_aio_client = gcs_aio_client
         assert _internal_kv_initialized()
 
-    async def put_info(self, job_id: str, job_info: JobInfo):
-        await self._gcs_aio_client.internal_kv_put(
+    async def put_info(
+        self, job_id: str, job_info: JobInfo, overwrite: bool = True
+    ) -> int:
+        """Put job info to the internal kv store.
+
+        Args:
+            job_id: The job id.
+            job_info: The job info.
+            overwrite: Whether to overwrite the existing job info.
+
+        Returns:
+            The number of keys added. If overwrite is True, this will be 1 if the
+                key was added and 0 if the key was updated. If overwrite is False,
+                this will be 1 if the key was added and 0 if the key already exists.
```

Comment:
should this just be a bool instead of a magic int?

Reply:
Thought about this, but I worry that the bool would be interpreted as "successful." The semantics are different from `num_added`

Reply:
Don't have a strong preference either way though, if you think the bool wouldn't be unexpected or weird I'll change it

Reply:
I see your point.

I think if you make it a bool and name it something like `new_key_added` in the docstring and callsite that's probably clearest, but also no strong preference. it just took me awhile to grok this docstring and a bool would've been easier